src/Models.js

```diff
@@ -72,6 +73,58 @@
         console.warn(`You are trying to finish work without starting one.`);
       }
     }
+
+    getWorkInfo() {
+      const type = this.history[this.startTime].type;
+      const timeLapsed = LifeGamification.utils.calcTime(this.startTime);
+      if(type.name === "normal"){
+        return {"type": type, "time": timeLapsed};
+      }
+      if(type.name === "countdown"){
+        const time = type.info.countdown * minute - timeLapsed;
+        return {"type": type, "time": time};
+      }
+      if(type.name === "pomodoro"){
+        let pomodorosFinished = 0;
```

Comment:
```
if(type.name === "pomodoro") {
  return pomodoroWorkInfo(some_arguments);
}
```
Thats what I meant when I said that this function is too long and too everything.